src/storage.rs

```diff
@@ -122,8 +125,13 @@ pub fn client_for_datastore(datastore: &DataStore) -> S3Client {
     S3Client::new_with(dispatcher, provider, region)
 }
 
-// <p>Function used to verify if a datastore is valid in terms of reachability</p>
-pub fn can_reach_datastore(datastore: &DataStore) -> bool {
+pub enum ReachableDatastoreError {
+    NoSuchBucket(String),
+    Unknown,
```

Comment:
Could we have a String added to the Unknown variant to indicate what the error the S3 backend returned?
Like, 
```suggestion
    Unknown(String),
```

Reply:
ready, I improved the way we wrap around the errors so it's more future ready